### PR TITLE
MLIBZ-1735: ping call

### PR DESCRIPTION
**Pull request by heyzooi**
#### Description

Adding missing ping call that we used to have in v1 of the library

#### Changes

* `ping()` method added to `Client`

#### Tests

* Unit Test to test successful and failure cases

### CODE REVIEW

File: Kinvey/Kinvey/Client.swift [hunk 1]
```diff
@@ -266,10 +267,62 @@ open class Client: Credential {
         return Client.fileURL(appKey: self.appKey!, tag: tag)
     }
     
-    public func encode(with aCoder: NSCoder) {
```
Comment (thomasconner) [inline Kinvey/Kinvey/Client.swift]: Why was this removed?

Comment (heyzooi) [inline Kinvey/Kinvey/Client.swift]: this is a legacy code when we had some support for Obj-C, so no need to keep it.

Comment (thomasconner) [inline Kinvey/Kinvey/Client.swift]: Ok
